plugin/trino-influxdb/src/main/java/io/trino/plugin/influxdb/QueryUtils.java

```diff
@@ -0,0 +1,236 @@
+            columns.stream().map(QueryUtils::wrapQuota).forEach(query::column);
+        }
+        else {
+            columnHandles.stream().map(col -> wrapQuota(col.getName())).forEach(query::column);
+        }
+    }
+
+    private static List<String> buildProjection(List<ColumnHandle> assignments)
+    {
+        if (assignments == null || assignments.isEmpty()) {
+            return ImmutableList.of();
+        }
+        ImmutableList.Builder<String> columns = ImmutableList.builder();
+        assignments.stream()
+                .filter(columnHandle -> columnHandle instanceof InfluxColumnHandle)
+                .map(InfluxColumnHandle.class::cast)
+                .map(InfluxColumnHandle::getName)
+                .forEach(columns::add);
+        return columns.build();
+    }
+
+    private static void appendWhereClause(SelectQueryImpl query, InfluxTableHandle tableHandle)
+    {
+        TupleDomain<ColumnHandle> constraint = tableHandle.getConstraint();
+        constraint.getDomains().ifPresent(columnHandleDomainMap -> {
+            WhereQueryImpl<SelectQueryImpl> where = query.where();
+            columnHandleDomainMap.forEach((columnHandle, domain) ->
+                    appendDomainFilter(where, (InfluxColumnHandle) columnHandle, domain));
+        });
+    }
+
+    private static void appendDomainFilter(WhereQueryImpl<SelectQueryImpl> where, InfluxColumnHandle columnHandle, Domain domain)
+    {
+        String columnName = wrapQuota(columnHandle.getName());
+        domain.getValues().getValuesProcessor().consume(
+                ranges -> appendDomainRangeFilter(where, columnName, ranges),
+                discreteValues -> appendDomainDiscreteFilter(where, columnName, discreteValues),
+                allOrNone -> {});
+    }
+
+    private static void appendDomainRangeFilter(WhereQueryImpl<SelectQueryImpl> where, String columnName, Ranges ranges)
+    {
+        List<Range> orderedRanges = ranges.getOrderedRanges();
+        WhereNested<WhereQueryImpl<SelectQueryImpl>> nestedWhere = where.andNested();
+        if (orderedRanges.stream().allMatch(x -> x.getType() instanceof TimestampType)) {
+            appendDomainTimeRangeFilter(nestedWhere, columnName, orderedRanges);
+        }
+        else if (orderedRanges.stream().allMatch(x -> x.getType() == DOUBLE || x.getType() == BIGINT)) {
+            appendDomainNumRangeFilter(nestedWhere, columnName, orderedRanges);
+        }
+        else if (orderedRanges.stream().allMatch(x -> x.getType() == BOOLEAN)) {
+            appendDomainBooleanRangeFilter(nestedWhere, columnName, orderedRanges);
+        }
+        else if (orderedRanges.stream().allMatch(x -> x.getType() == VARCHAR)) {
+            appendDomainVarcharRangeFilter(nestedWhere, columnName, orderedRanges);
+        }
+        nestedWhere.close();
+    }
+
+    private static void appendDomainTimeRangeFilter(WhereNested<WhereQueryImpl<SelectQueryImpl>> where, String columnName, List<Range> ranges)
```

Comment:
Can we merge with `appendDomainNumRangeFilter` as the only difference we see is how we process the values.